services/pipeline/src/main/java/tech/pegasys/pantheon/services/pipeline/Pipeline.java

```diff
@@ -49,7 +49,7 @@
 
   Pipeline(
       final Pipe<?> inputPipe,
-      final Collection<Runnable> stages,
+      final Collection<Stage> stages,
```

Comment:
Why Collection?  Does the order of the stages matter?  That would imply List.

Reply:
It doesn't matter at this point - we're going to execute them all in their own threads anyway.  The order of stages is really enforced by the way the pipes connect them.